spec/unit/provider/osx_profile_spec.rb

```diff
@@ -0,0 +1,251 @@
+      allow(@provider).to receive(:get_new_profile_hash).and_return(@test_profile)
+      allow(@provider).to receive(:get_installed_profiles).and_return(@all_profiles)
+      allow(@provider).to receive(:read_plist).and_return(@all_profiles)
+      allow(::File).to receive(:unlink).and_return(true)
+    end
+
+    it 'should build the get all profiles shellout command correctly' do
+      profile_name = 'com.testprofile.screensaver.mobileconfig'
+      tempfile = '/tmp/allprofiles.plist'
+      @new_resource.profile_name profile_name
+      allow(@provider).to receive(:generate_tempfile).and_return(tempfile)
+      allow(@provider).to receive(:get_installed_profiles).and_call_original
+      allow(@provider).to receive(:read_plist).and_return(@all_profiles)
+      expect(@provider).to receive(:shell_out!).with("profiles -P -o '/tmp/allprofiles.plist'")
+      @provider.load_current_resource
+    end
+
+    it 'should use profile name as profile when no profile is set' do
+      profile_name = 'com.testprofile.screensaver.mobileconfig'
+      @new_resource.profile_name profile_name
+      @provider.load_current_resource
+      expect(@new_resource.profile_name).to eql(profile_name)
+    end
+
+    it 'should use identifier from specified profile' do
+      @new_resource.profile @test_profile
+      @provider.load_current_resource
+      expect(
+        @provider.instance_variable_get(:@new_profile_identifier)
+        ).to eql(@test_profile['PayloadIdentifier'])
+    end
+
+    it 'should install when not installed' do
+      @new_resource.profile @test_profile
+      allow(@provider).to receive(:get_installed_profiles).and_return(@no_profiles)
+      @provider.load_current_resource
+      expect { @provider.run_action(:install) }
+    end
+
+    it 'should install when installed but uuid differs' do
+      @new_resource.profile @test_profile
+      @all_profiles['_computerlevel'][1]['ProfileUUID'] = '1781fbec-3325-565f-9022-9bb39245d4dd'
+      @provider.load_current_resource
+      expect { @provider.run_action(:install) }
+    end
+
+    it 'should build the shellout install command correctly' do
+      profile_path = '/tmp/test.mobileconfig'
+      @new_resource.profile @test_profile
+      # Change the profile so it triggers an install
+      @all_profiles['_computerlevel'][1]['ProfileUUID'] = '1781fbec-3325-565f-9022-9bb39245d4dd'
+      @provider.load_current_resource
+      allow(@provider).to receive(:write_profile_to_disk).and_return(profile_path)
+      expect(@provider).to receive(:shell_out).with("profiles -I -F '#{profile_path}'").and_return(shell_out_success)
+      @provider.action_install()
+    end
+
+    it 'should fail if there is no identifier inside the profile' do
+      @test_profile.delete('PayloadIdentifier')
+      @new_resource.profile @test_profile
+      expect{@provider.run_action(:install)}.to raise_error(RuntimeError)
```

Comment:
Because `RuntimeError` is so broad I think you should check the error message here - `... raise_error(RuntimeError, /message/)`